lib/pbench/test/unit/server/test_upload.py

```diff
@@ -252,6 +260,46 @@ def test_bad_extension_upload(
         self.verify_logs(caplog)
         assert not self.cachemanager_created
 
+    @pytest.mark.parametrize("error", (errno.ENOSPC, errno.ENFILE, None))
+    def test_bad_stream_read(
+        self, client, server_config, pbench_drb_token, monkeypatch, error
+    ):
+        """Test handling of errors from the intake stream read
+
+        The intake code handles errno.ENOSPC specially; however although the
+        code tried to raise an APIAbort with HTTPStatus.INSUFFICIENT_SPACE
+        (50), the werkzeug abort() doesn't support this and ends up with
+        a generic internal server error. Instead, we now have three distinct
+        cases which all result (to the client) in identical internal server
+        errors. Nevertheless, we exercise all three cases here.
+        """
+        stream = BytesIO(b"12345")
+
+        def access(self, intake: Intake, request: Request) -> Access:
+            return Access(5, stream)
+
+        def read(self):
+            if error:
+                e = OSError()
+                e.errno = error
+            else:
+                e = Exception("Nobody expects the Spanish Exception")
+            raise e
+
+        monkeypatch.setattr(Upload, "_access", access)
+        monkeypatch.setattr(stream, "read", read)
```

Comment:
This is an interesting cheat:  you have the _read_ operation reporting `errno.ENOSPC`...when does that ever happen?  🤣  Shouldn't that be the `write()` call behavior?  😇

Reply:
Sure, it's a cheat because I don't have a handle on the class write stream, and this seemed like a reasonable proxy as its in the same loop. (Java would complain about raising an undeclared exception, but Python doesn't care. 😸 )